lib/hydra/pcdm/deep_member_iterator.rb

```diff
@@ -0,0 +1,25 @@
+module Hydra::PCDM
+  class DeepMemberIterator
+    include Enumerable
+    attr_reader :record
+    def initialize(record)
+      @record = HasMembers.new(record)
+    end
+
+    def each
+      record.members.each do |member|
+        yield member
+      end
+      record.members.each do |member|
```

Comment:
I have this rough idea that this'll be slightly more performant because of the average case, but I'd be happy to go deep instead.